tests/socket.io-client.ts

```diff
@@ -10,5 +10,5 @@ import { Socket } from 'socket.io-client'
 import { io } from 'socket.io-client/dist/socket.io.js'
 
 export function createSocketClient(uri: string): Socket {
-  return io(uri, { transports: ['websocket'] })
+  return io(uri, { transports: ['websocket'], forceNew: true })
```

Comment:
What does `forceNew` do? 👀

Reply:
> Because our tests now create sockets on multiple namespaces, Engine.IO will multiplex them across a single WebSocket. This means we're leaking state between test cases, and only one test's interceptor connection handler will fire. Resolve this by disabling shared WebSockets.

I could add a comment to this effect if desired.